glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -262,6 +276,32 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
      */
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun getCalendarInstance(): Calendar = Calendar.getInstance()
+
+    /**
+     * Update flag to show we are no longer in the foreground.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onEnterBackground() {
+        isInForeground = false
+    }
+
+    /**
+     * Update the flag to indicate we are moving to the foreground, and if Glean is initialized we
+     * will check to see if the metrics ping needs scheduled for collection.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    fun onEnterForeground() {
+        isInForeground = true
+
+        // We check for the metrics ping schedule here because the app could have been in the
+        // background and resumed in which case Glean would already be initialized but we still need
+        // to perform the check to determine whether or not to collect and schedule the metrics ping.
+        // If this is the first ON_START event since the app was launched, Glean wouldn't be
+        // initialized yet.
+        if (Glean.isInitialized()) {
+            Glean.metricsPingScheduler.schedule()
```

Comment:
Isn't this essentially `this.schedule()` or is there any difference referencing it through the `Glean.metricsPingScheduler`?

Reply:
Yes, good point!  It is my fault for not changing this when I moved it from the `GleanLifecycleObserver`.